src/dos/dos_devices.cpp

```diff
@@ -420,6 +430,93 @@ void DOS_DelDevice(DOS_Device * dev) {
 	}
 }
 
+// Structure of internal DOS tables, Device Driver
+//
+// Property   Offset  Type      Description
+// ~~~~~~~~   ~~~~~~  ~~~~~     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// next_rpt   00h     DWord --> Next driver in chain (x:FFFF means end)
+// attributes 04h     Word      Device attributes.
+// strategy   06h     Word      Device strategy routine offset.
+// interrupt  08h     Word      Device interrupt routine offset.
+// name       0Ah     8 Bytes   Device name padded with spaces.
+//
+// Ref: https://www.infradeaorg/devload/DOSTables.html, Appendix 2
+//
+// Indicates if the device at the given real pointer is the last in DOS's chain
+bool DOS_IsLastDevice(const RealPt rp)
+{
+	constexpr uint16_t last_offset_marker = 0xffff;
+	return RealOff(rp) == last_offset_marker;
+}
+
+// From the given real pointer, get the next device driver's real pointer
+RealPt DOS_GetNextDevice(const RealPt rp)
+{
+	constexpr auto next_pos = 0x00; // see dev structure table above
+	return real_readd(RealSeg(rp), RealOff(rp) + next_pos);
+}
+
+// Get the tail real pointer from the DOS device driver linked list
+RealPt DOS_GetLastDevice()
+{
+	auto rp = dos_infoblock.GetDeviceChain();
+
+	while (!DOS_IsLastDevice(rp)) {
+		rp = DOS_GetNextDevice(rp);
+	}
+	return rp;
+}
+
+// Append the device at the given segment to the end of the DOS device linked list
+void DOS_AppendDevice(const uint16_t new_segment, const uint16_t new_offset)
+{
+	const auto new_rp  = RealMake(new_segment, new_offset);
+	const auto tail_rp = DOS_GetLastDevice();
+	real_writed(RealSeg(tail_rp), RealOff(tail_rp), new_rp);
+}
+
+bool DOS_DeviceHasAttributes(const RealPt rp, const uint16_t req_attributes)
+{
+	constexpr auto attributes_pos = 0x04; // see dev structure table above
+	const auto attributes = real_readw(RealSeg(rp), RealOff(rp) + attributes_pos);
+	return (attributes & req_attributes) == req_attributes;
+}
+
+uint16_t DOS_GetDeviceStrategy(const RealPt rp)
+{
+	constexpr auto strategy_pos = 0x06; // see dev structure table above
+	return real_readw(RealSeg(rp), RealOff(rp) + strategy_pos);
+}
+
+uint16_t DOS_GetDeviceInterrupt(const RealPt rp)
+{
+	constexpr auto interrupt_pos = 0x08; // see dev structure table above
+	return real_readw(RealSeg(rp), RealOff(rp) + interrupt_pos);
+}
+
+bool DOS_DeviceHasName(const RealPt rp, const std::string_view req_name)
+{
+	constexpr auto name_pos   = 0x0a; // see dev structure table above
+	constexpr size_t name_len = 8;    // see dev structure table above
```

Comment:
Would it be an improvement to turn these into namespaced constants defined at a single location, which would obviate the need for these comments at every single usage?

E.g.

```cpp
// Structure of internal DOS tables, Device Driver
// Ref: https://www.infradeaorg/devload/DOSTables.html, Appendix 2

namespace DeviceDriverInfo {

constexpr uint8_t next_rpt_offset = 00h; // DWORD, points to next driver in
                                         // chain (x:FFFF means end)
constexpr uint8_t attributes_offset = 04h; // WORD, device attributes offset
constexpr uint8_t strategy_offset   = 06h; // WORD, device strategy routine
                                           // offset.
constexpr uint8_t interrupt_offset = 08h;  // WORD, device interrupt
                                           // routine offset.
constexpr uint8_t name_offset = 0ah;       // 8 bytes, device name padded with
                                           // spaces.
} // namespace DeviceDriverInfo
```

Reply:
Added right under the comment block; easier to reference and more consistent in the functions - thanks, @johnnovak !